src/gl/lineFeature.js

```diff
@@ -123,162 +123,107 @@ geo.gl.lineFeature = function (arg) {
-        start = null,
+    var data = m_this.data(),
+        i, j, k, v,
+        numSegments = 0,
+        lineItem, lineItemData,
+        vert = [{}, {}], vertTemp,
+        pos, posIdx3,
         position = [],
-        strkWidthArr = [],
-        strkColorArr = [],
-        strkOpacityArr = [],
-        geom = vgl.geometryData(),
         posFunc = m_this.position(),
         strkWidthFunc = m_this.style.get('strokeWidth'),
         strkColorFunc = m_this.style.get('strokeColor'),
         strkOpacityFunc = m_this.style.get('strokeOpacity'),
-        buffers = vgl.DataBuffers(1024),
-        // Sources
-        posData = vgl.sourceDataP3fv({'name': 'pos'}),
-        prvPosData = vgl.sourceDataAnyfv(
-            3, vgl.vertexAttributeKeysIndexed.Four, {'name': 'prev'}),
-        nxtPosData = vgl.sourceDataAnyfv(
-            3, vgl.vertexAttributeKeysIndexed.Five, {'name': 'next'}),
-        offPosData = vgl.sourceDataAnyfv(
-            1, vgl.vertexAttributeKeysIndexed.Six, {'name': 'offset'}),
-        strkWidthData = vgl.sourceDataAnyfv(
-            1, vgl.vertexAttributeKeysIndexed.One, {'name': 'strokeWidth'}),
-        strkColorData = vgl.sourceDataAnyfv(
-            3, vgl.vertexAttributeKeysIndexed.Two, {'name': 'strokeColor'}),
-        strkOpacityData = vgl.sourceDataAnyfv(
-            1, vgl.vertexAttributeKeysIndexed.Three,
-            {'name': 'strokeOpacity'}),
-        // Primitive indices
-        triangles = vgl.triangles(),
         order = m_this.featureVertices(),
-        addVert = function (prevPos, currPos, nextPos, offset,
-                            width, color, opacity) {
-          buffers.write('prev', prevPos, currIndex, 1);
-          buffers.write('pos', currPos, currIndex, 1);
-          buffers.write('next', nextPos, currIndex, 1);
-          buffers.write('offset', [offset], currIndex, 1);
-          buffers.write('indices', [currIndex], currIndex, 1);
-          buffers.write('strokeWidth', [width], currIndex, 1);
-          buffers.write('strokeColor', color, currIndex, 1);
-          buffers.write('strokeOpacity', [opacity], currIndex, 1);
-          currIndex += 1;
-        };
-
-    m_this.data().forEach(function (item) {
-      lineItem = m_this.line()(item, itemIndex);
-      lineSegments.push(lineItem.length);
-      lineItem.forEach(function (lineItemData) {
-        pos = posFunc(lineItemData, lineItemIndex, item, itemIndex);
+        buffers,
+        posBuf, nextBuf, prevBuf, offsetBuf, indicesBuf,
+        strokeWidthBuf, strokeColorBuf, strokeOpacityBuf,
+        dest, dest3,
+        geom = m_mapper.geometryData();
+
+    for (i = 0; i < data.length; i += 1) {
+      lineItem = m_this.line()(data[i], i);
+      numSegments += lineItem.length - 1;
+      for (j = 0; j < lineItem.length; j += 1) {
+        pos = posFunc(lineItem[j], j, lineItem, i);
         if (pos instanceof geo.latlng) {
-          position.push([pos.x(), pos.y(), 0.0]);
+          position.push(pos.x());
```

Comment:
Two reasons: (1) the transform is faster on a single array than on an array of arrays, and (2) less memory use.  A single array uses less memory than an array of arrays.  Also, with the array or arrays we end up creating one array per point initially, then one array per point in the transform.